internal/car/car.go

```diff
@@ -84,14 +88,72 @@ func (c *car) List(ctx context.Context, tag, platform string) error {
 			break
 		}
 	}
-
 	unmatched := pm.Unmatched()
 	if len(unmatched) > 0 {
 		return fmt.Errorf("%s not found in layer", strings.Join(unmatched, ", "))
 	}
 	return nil
 }
 
+func (c *car) Extract(ctx context.Context, tag, platform, directory string, stripComponents int) error {
+	// maintain a lazy map of directories already created
+	dirsCreated := map[string]bool{}
+	return c.do(ctx, func(name string, size int64, mode os.FileMode, modTime time.Time, reader io.Reader) error {
+		destinationPath, ok := newDestinationPath(name, directory, stripComponents)
+		if !ok {
+			return nil // skip
+		}
+
+		baseDir := path.Dir(destinationPath)
+		if ok := dirsCreated[baseDir]; !ok {
+			if err := os.MkdirAll(baseDir, 0755); err != nil { //nolint:gosec
+				return err
+			}
+			dirsCreated[baseDir] = true
+		}
+		fw, err := os.OpenFile(destinationPath, os.O_CREATE|os.O_RDWR, mode) //nolint:gosec
+		if err != nil {
+			return err
+		}
+		if c.veryVerbose {
+			c.listVerbose(name, size, mode, modTime)
+		} else if c.verbose {
+			fmt.Fprintln(c.out, name)
+		}
+		_, err = io.CopyN(fw, reader, size)
+		return err
+	}, tag, platform)
+}
+
+func newDestinationPath(name, directory string, stripComponents int) (string, bool) {
```

Comment:
I think we should put comment for this function e.g. adding // strip the original path by stripComponents

Reply:
for the future!

Reply:
sgtm